### PR TITLE
New example: non-orographic gravity waves on a small planet

### PR DESCRIPTION
**Pull request by thomasgibson**
Gusto should have some 3D examples, so I've added the DCMIP test case 3-1 (see title). This uses the recently added hybridized compressible solver.

### CODE REVIEW

**Comment by jshipton** [comment]
Fantastic! Thanks for setting this up! Interested to see how much time it adds to the testing...

**Comment by thomasgibson** [comment]
I tried to set the problem up so that it's not *too* demanding for a laptop. It's very coarse. I also configured the solver options (for the balancing calculation during initialization) and it's a bit faster than the default parameters.

We still need to work on a better solver for the balanced density computation. I think hybridizing vertically in the columns may be a potential solution. This will require some work though, as I don't think we have the right functionality to create a "vertical" trace space.

File: examples/dcmip_3_1_meanflow_quads.py [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+mesh = ExtrudedMesh(m, layers=nlayers,
+                    layer_height=z_top/nlayers,
+                    extrusion_type="radial")
+
+# Create polar coordinates:
+# Since we use a CG1 field, this is constant on layers
+W_Q1 = FunctionSpace(mesh, "CG", 1)
+z_expr = Expression("sqrt(x[0]*x[0] + x[1]*x[1] + x[2]*x[2]) - a", a=a)
+z = Function(W_Q1).interpolate(z_expr)
+lat_expr = Expression("asin(x[2]/sqrt(x[0]*x[0] + x[1]*x[1] + x[2]*x[2]))")
```
Comment (tommbendall) [inline examples/dcmip_3_1_meanflow_quads.py]: Does this need to be a c expression or can you write it with Firedrake functions?

Comment (thomasgibson) [reply examples/dcmip_3_1_meanflow_quads.py]: No, you are right...thanks for holding me accountable :P I will update.

**Pull request merged**